io-engine/tests/resource_stats.rs

```diff
@@ -0,0 +1,408 @@
+    let repl_1_stat: &Vec<ReplicaIoStats> = &repl_1_stat.get_ref().stats;
+    let repl_2_stat: &Vec<ReplicaIoStats> = &repl_2_stat.get_ref().stats;
+    let repl_0_stat: &Vec<ReplicaIoStats> = &repl_0_stat.get_ref().stats;
+    let nexus_stat = nexus_stat.get(0).unwrap();
+    let pool_0_stat = pool_0_stat.get(0).unwrap();
+    let pool_1_stat = pool_1_stat.get(0).unwrap();
+    let pool_2_stat = pool_2_stat.get(0).unwrap();
+    let repl_1_stat = repl_1_stat.get(0).unwrap();
+    let repl_2_stat = repl_2_stat.get(0).unwrap();
+    let repl_0_stat = repl_0_stat.get(0).unwrap();
+
+    // Validate non zero num_write_ops.
+    assert_ne!(nexus_stat.num_write_ops, 0);
+    assert_ne!(pool_0_stat.num_write_ops, 0);
+    assert_ne!(pool_1_stat.num_write_ops, 0);
+    assert_ne!(repl_1_stat.stats.clone().unwrap().num_write_ops, 0);
+    assert_ne!(repl_2_stat.stats.clone().unwrap().num_write_ops, 0);
+    assert_ne!(repl_0_stat.stats.clone().unwrap().num_write_ops, 0);
+
+    // Validate num_read_ops of nexus and replica.
+    let replica_num_read_ops = repl_1_stat.stats.clone().unwrap().num_read_ops
+        + repl_2_stat.stats.clone().unwrap().num_read_ops
+        + repl_0_stat.stats.clone().unwrap().num_read_ops;
+    assert_eq!(nexus_stat.num_read_ops, replica_num_read_ops);
+
+    // Validate num_read_ops of pool and replica.
+    assert_eq!(
+        repl_1_stat.stats.clone().unwrap().num_read_ops,
+        pool_0_stat.num_read_ops
+    );
+    assert_eq!(
+        repl_2_stat.stats.clone().unwrap().num_read_ops,
+        pool_1_stat.num_read_ops
+    );
+    assert_eq!(
+        repl_0_stat.stats.clone().unwrap().num_read_ops,
+        pool_2_stat.num_read_ops
+    );
+
+    // Validate num_write_ops of replica and pool.
+    assert_eq!(
+        repl_1_stat.stats.clone().unwrap().num_write_ops,
+        pool_0_stat.num_write_ops
+    );
+    assert_eq!(
+        repl_2_stat.stats.clone().unwrap().num_write_ops,
+        pool_1_stat.num_write_ops
+    );
+    assert_eq!(
+        repl_0_stat.stats.clone().unwrap().num_write_ops,
+        pool_2_stat.num_write_ops
+    );
+
+    // Validate num_write_ops of nexus and replica.
+    assert_eq!(
+        nexus_stat.num_write_ops,
+        repl_1_stat.stats.clone().unwrap().num_write_ops
+    );
+
+    // Validate num_write_ops of replica_0 and replica_1.
+    assert_eq!(
+        repl_1_stat.stats.clone().unwrap().num_write_ops,
+        repl_2_stat.stats.clone().unwrap().num_write_ops
+    );
+
+    // Validate num_write_ops of pool_0, pool_2 and pool_1.
+    assert_eq!(pool_0_stat.num_write_ops, pool_1_stat.num_write_ops);
+    assert_eq!(pool_1_stat.num_write_ops, pool_2_stat.num_write_ops);
+
+    // Validate write_latency_ticks of replica and pool
+    if repl_1_stat.stats.clone().unwrap().write_latency_ticks
+        <= pool_0_stat.write_latency_ticks
+    {
+        panic!("replica wr latency less then pool wr latency");
+    }
```

Comment:
Should the assert hit for `<=` or only `<`
```suggestion
assert!(
    repl_1_stat.stats.clone().unwrap().write_latency_ticks > pool_0_stat.write_latency_ticks,
    "replica wr latency less then pool wr latency"
);
```

Reply:
They cant be equal also. Soshould fail if replica_latency <= pool_latency